testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/ExpectedDeprecationWarning.java

```diff
@@ -95,6 +98,11 @@ public boolean matchesNextLines(List<String> lines, int startIndex) {
         String nextLines = numLines == 1
             ? lines.get(startIndex)
             : String.join("\n", lines.subList(startIndex, Math.min(startIndex + numLines, lines.size())));
+
+        // When info or debug logging is enabled, the line will be prefixed with the log level and timestamp.
+        // We need to strip this out to match the expected message.
+        nextLines = DEPRECATION_WARNING_LOG_PREFIX_PATTERN.matcher(nextLines).replaceAll("");
```

Comment:
🤔 Shouldn't we _only_ do this stripping if debug logging is enabled?

Reply:
Probably, though it is more difficult than you think to detect which log level we’re at.

considering you can set a system property, cli arg, Gradle property (via -P or via gradle.properties) it gets complicated fast

Reply:
Didn't we have an API for figuring out if something was enabled regardless of where it was enabled from?  I'm drawing a blank.

Reply:
Maybe in the build process itself? But this code runs after execution is finished